lib/types/index.js

```diff
@@ -291,16 +291,38 @@ export type RetrievalMethod = $Values<typeof RETRIEVAL_METHODS>
 
 export type FetchFrequency = $Values<typeof FETCH_FREQUENCIES>
 
-export type FeedTransformation = {
+export type Substitution = {
+  normalizeSpace?: boolean,
+  pattern: string,
+  replacement: string
+}
+
+export type NormalizeFieldFields = {
+  capitalize?: boolean,
+  capitalizeExceptions?: string,
+  fieldName?: string,
+  performSubstitutions?: boolean,
+  substitutions?: Array<Substitution>
+}
+
+export type ReplaceFileFromStringFields = {
+  csvData?: ?string
+}
+
+export type FeedTransformationBase = {
   '@type': $Values<typeof FEED_TRANSFORMATION_TYPES>,
   active: boolean,
-  csvData?: string,
   matchField?: string,
   matchValues?: Array<string>,
   sourceVersionId?: string,
   table: ?string
 }
 
+export type FeedTransformation =
+  & FeedTransformationBase
+  & ReplaceFileFromStringFields
+  & NormalizeFieldFields
```

Comment:
I'd prefer if each transformation type extended the `FeedTransformationBase` and then this `FeedTransformation` type would be a [union type](https://flow.org/en/docs/types/unions/) of all the possible transformations.

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/649/commits/54f1aac3573cc5af4575bd839afb116708e626e8. There is one prop `typeName` I had to add that duplicates `@type`, without which flow rejects the case in `getTransformationName` that involves `csvData`. (Flow does not seem to accept attribute `@type` for type branching)